### DISCUSSION TITLE
Please try out the release candidate in your projects and let us know if you encounter issues!

### DISCUSSION THREAD

**Discussion by c-bata** [optuna/optuna#3864] [Announcements]
As always, thanks for all the feedback and contributions. We've just published the release candidate of the upcoming major update v3. We plan to release v3.0.0 within a few weeks. Please try out the release candidate to ensure Optuna v3.0.0 works in your projects and help us gather feedback early!

```
$ pip install optuna==3.0.0rc0
```

### How to Report Bugs

If you encounter issues, please report them by filing GitHub issues or responding to this thread.

https://github.com/optuna/optuna/issues/new?assignees=&labels=bug&template=bug-report.yml

### Release Notes

* [v3.0.0-rc0](https://github.com/optuna/optuna/releases/tag/v3.0.0-rc0)
* [v3.0.0-b1](https://github.com/optuna/optuna/releases/tag/v3.0.0-b1)
* [v3.0.0-b0](https://github.com/optuna/optuna/releases/tag/v3.0.0-b0)
* [v3.0.0-a2](https://github.com/optuna/optuna/releases/tag/v3.0.0-a2)
* [v3.0.0-a1](https://github.com/optuna/optuna/releases/tag/v3.0.0-a1)
* [v3.0.0-a0](https://github.com/optuna/optuna/releases/tag/v3.0.0-a0)

**Comment by nzw0301** [comment]
As reported by https://github.com/optuna/optuna/issues/3558#issuecomment-1225012202,
if we upgrade Optuna to v3.0.0rc rather than install optuna from scratch, `typing_extensions` might be old for Optuna v3.0.0rc, where we use a newer class `typing_extensions.ParamSpec`. Shall we introduce a version constraint for `typing_extensions` to avoid this error?

**Reply by c-bata** [reply]
Thanks for letting me know!
Sounds good 👍  Could you open a pull request if you have time?

**Reply by nzw0301** [reply]
Sure!

**Comment by c-bata** [comment]
I've just posted the Optuna v3 migration guide on GitHub Discussions. It would be great if you check it out and try out the release candidate in your projects.

[Optuna v3 Migration Guide](https://github.com/optuna/optuna/discussions/3930)
